### PR TITLE
Formalize type concept and add 'map' and 'any' types.

### LINKED ISSUE #74: Map support
Declared as `map<string, string>`

### LINKED ISSUE #79: Any type support

### PR DESCRIPTION
**Pull request by JakeWharton**
Closes #74. Closes #79.

### CODE REVIEW

File: src/main/java/com/squareup/protoparser/ProtoParser.java [hunk 1]
```diff
@@ -648,6 +651,54 @@ private String readName() {
     return optionName;
   }
 
+  /** Reads a scalar, map, or type name. */
+  private DataType readDataType() {
+    String name = readWord();
+    switch (name) {
```
Comment (JakeWharton) [inline src/main/java/com/squareup/protoparser/ProtoParser.java]: https://twitter.com/JakeWharton/status/558000564978855936

**Comment by JakeWharton** [comment]
Next: port Wire to 4.0.0-SNAPSHOT to flesh out bugs and then ship!

**Comment by zachmargolis** [comment]
LGTM

File: src/main/java/com/squareup/protoparser/DataType.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
+// Copyright 2015 Square, Inc.
```
Comment (swankjesse) [inline src/main/java/com/squareup/protoparser/DataType.java]: off topic: big headers?

File: src/test/java/com/squareup/protoparser/ProtoParserTest.java [hunk 3]
```diff
@@ -725,6 +821,29 @@
...
+  @Test public void serviceTypesMustBeNamed() {
+    try {
+      String proto = ""
+          + "service SearchService {\n"
+          + "  rpc Search (string) returns (SearchResponse);"
+          + "}";
+      ProtoParser.parse("test.proto", proto);
+      fail();
+    } catch (IllegalStateException e) {
+      assertThat(e).hasMessage("Syntax error in test.proto at 2:21: expected message but was string");
```
Comment (swankjesse) [inline src/test/java/com/squareup/protoparser/ProtoParserTest.java]: nice

**Pull request merged**